tests/testthat/test-utilities-pk-analysis.R

```diff
@@ -23,6 +23,17 @@ test_that("It should return null when retrieving a pk parameter that does not ex
   expect_null(pkParameter)
 })
 
+test_that("it can return pk analysis for sparse population and only return ids defined in the population file", {
+  clearOutputs(sim)
+  addOutputs(outputs, sim)
+  population <- loadPopulation(getTestDataFilePath("pop_5_spared_id.csv"))
+  results <- runSimulation(sim, population = population)
+  pkAnalyses <- calculatePKAnalyses(results = results)
+  df <- pkAnalysesToDataFrame(pkAnalyses)
+  d <- dplyr::distinct(df, IndividualId)
+  expect_equal(nrow(d), 5)
```

Comment:
Checking that there are only 5 different values in the actual df. Before it would have been 21

Reply:
I don't know the context, but do you think it makes sense for `pkAnalysesToDataFrame()` to return only distinct rows?

So instead of the user calling `dplyr::distinct()`, like here, the function would handle this internally.

Reply:
No no. This is not what I am trying to do. This is a test and I want to verify that I only have 5 unique values in one column of the returned table. (This is the IndividualId) column. But the values can definitely repeat themselves, one for each pk parameter